dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/EPersonRestRepository.java

```diff
@@ -65,6 +126,78 @@ public EPersonRest findOne(Context context, UUID id) {
         return page;
     }
 
+    /**
+     * Find the epersons matching the query q parameter. The search is delegated to the
+     * {@link EPersonService#search(Context, String, int, int)} method
+     *
+     * @param q
+     *            is the *required* query string
+     * @param pageable
+     *            contains the pagination information
+     * @return a Page of EPersonRest instances matching the user query
+     */
+    @SearchRestMethod(name = "byName")
+    public Page<EPersonRest> findByName(@Parameter(value = "q", required = true) String q,
+            Pageable pageable) {
+        List<EPerson> epersons = null;
+        int total = 0;
+        try {
+            Context context = obtainContext();
+            epersons = es.search(context, q, pageable.getOffset(), pageable.getOffset() + pageable.getPageSize());
+            total = es.searchResultCount(context, q);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        Page<EPersonRest> page = new PageImpl<EPerson>(epersons, pageable, total).map(converter);
+        return page;
+    }
+
+    /**
+     * Find the eperson with the provided email address if any. The search is delegated to the
+     * {@link EPersonService#findByEmail(Context, String)} method
+     *
+     * @param email
+     *            is the *required* email address
+     * @param pageable
+     *            contains the pagination information
+     * @return a Page of EPersonRest instances matching the user query
+     */
+    @SearchRestMethod(name = "byEmail")
+    public EPersonRest findByEmail(@Parameter(value = "email", required = true) String email) {
+        EPerson eperson = null;
+        try {
+            Context context = obtainContext();
+            eperson = es.findByEmail(context, email);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        if (eperson == null) {
+            return null;
+        }
+        return converter.fromModel(eperson);
+    }
+
+    @Override
+    protected void delete(Context context, UUID id) throws AuthorizeException, RepositoryMethodNotImplementedException {
```

Comment:
The signature line identifies 2 exceptions that are not actually thrown.  Will this compile without those declarations?
Interestingly, the meaningful exceptions are the runtime exceptions which do not need to be declared.

Reply:
@mwoodiupui had similar questions to this on #rest-api Slack today.  We both noticed that most of the `*RestRepository` classes (including this new one) throw a lot of `RuntimeException`s instead of clearly defining separate exceptions.  Seems like we should have a discussion about best practices with regards to exceptions in these classes.